doc/ConnectionPooling.md

```diff
@@ -0,0 +1,387 @@
+For that a [SnowflakeDbSessionPool.cs](/Snowflake.Data/Client/SnowflakeDbSessionPool.cs) is provided by
+- [SnowflakeDbSessionPool.GetPool(connectionString)](https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Client/SnowflakeDbConnectionPool.cs#L45)
+- [SnowflakeDbSessionPool.GetPool(connectionString, securePassword)](https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Client/SnowflakeDbConnectionPool.cs#L51).
+to control pool settings from the code. Changed pool settings are not reflected by their connection string therefore recommended way is to control the pool from the connection string.
+
+### Pool Lifecycle
+
+Single pool is instantiated each time an application creates and opens a connection for the first time using particular connection string.
+From that moment the pool tracks and maintains connections matching exactly this connection string.
+Pool is responsible for destroying and recreating connections which are old enough (see [Expiration Timeout](#expiration-timeout)).
+Number of connections is maintained within [Minimum pool size](#minimum-pool-size) and [Maximum pool size](#maximum-pool-size).
+Connections in all their statuses are tracked:
+- opening phase
+- busy phase
+- closed and returned to the pool (idle)
+User can clean up the pool using methods: [Clear Pool](#clear-pool).
+
+### Connection Lifecycle
+
+#### Opening
+
+Before connection is opened it is `scheduled` to be opened in the pool on a waiting queue. Connections scheduled to open are counted to the pool size.
+After connection is successfully opened it is counted as a `busy` in a separate list. Pool can provide already opened `idle` connection or if there are none it will schedule new one to open.
+When [Maximum pool size](#maximum-pool-size) is reached connection is waiting to be opened within period of time controlled with [Pool Size Exceeded Timeout](#pool-size-exceeded-timeout).
+When the timeout is exceeded then an exception will get thrown.
+
+#### Busy
+
+`Busy` connection is provided by the pool but it is counted to the pool size. It is returned to be reused during Close operation.
+When application does not Close connection it may hit the limit of [Maximum pool size](#maximum-pool-size).
+
+#### Closing
+
+When application closes the connection couple of things happen:
+- Pending transactions will be rolled back (if any)
+- Connection can be pooled when its properties are not changed
+- Connection with changed: database, schema, warehouse or role can be:
+  - pooled when OriginalPool mode enabled, see more: [Changed Session Behavior](#changed-session-behavior)
+  - destroyed when Destroy mode is set
+
+#### Evicting Connection
+
+Application may mark the connection to evict without turning off the pool. When connection Close methods gets executed it will not be pooled and if necessary
+a new connection will be created to maintain Minimum Pool Size.
```

Comment:
I would also write about `poolingEnabled=false` which is a more natural way to achieve that.